### ISSUE TITLE
BERT performance slower with Intel(R) Xeon(R)

### ISSUE THREAD

**Issue by xiaoerqi** [microsoft/onnxruntime#3076]
Describe the bug
I have exported a BERT model from tensorflow's transformers models.

I wrote a sample with C + + API.It works well on my personal computer(Intel(R) i9 9900k), but it's very very very slow on the company's servers(Intel(R) Xeon(R) Gold 5118)

Batch size: 1
personal computer: 13.028000 ms
company's servers: 599.910000 ms

Urgency

System information

OS Platform and Distribution: Ubuntu 14.04.5 LTS
CPU:Intel(R) Xeon(R) Gold 5118 CPU @ 2.30GHz
ONNX Runtime installed from: source(./build.sh --config Release --build_shared_lib --parallel --use_dnnl)
ONNX Runtime version: 1.1.0
Python version: 3.6.10
Visual Studio version (if applicable):
GCC/Compiler version (GCC/G++:4.8.4):
CUDA/cuDNN version: CPU only
GPU model and memory:

To Reproduce

  Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "test");
  Ort::SessionOptions session_options;
  session_options.SetIntraOpNumThreads(1);
  session_options.SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_ALL);
  OrtSessionOptionsAppendExecutionProvider_Dnnl(session_options, 1);
#ifdef _WIN32
  const wchar_t* model_path = L"/data/qizhang19/tools/test_onnxruntime/test_onnx/model/bertmodel.onnx";
#else
  const char* model_path = "/data/qizhang19/tools/test_onnxruntime/test_onnx/model/bertmodel.onnx";
#endif

  printf("Using Onnxruntime C++ API\n");
  Ort::Session session(env, model_path, session_options);
  Ort::AllocatorWithDefaultOptions allocator;

  // print number of model input nodes
  size_t num_input_nodes = session.GetInputCount();
  std::vector<const char*> input_node_names(num_input_nodes);
  std::vector<int64_t> input_node_dims;  // simplify... this model has only 1 input node {1, 3, 224, 224}.
                                         // Otherwise need vector<vector<>>

  printf("Number of inputs = %zu\n", num_input_nodes);

  for (int i = 0; i < num_input_nodes; i++) {
    // print input node names
    char* input_name = session.GetInputName(i, allocator);
    printf("Input %d : name=%s\n", i, input_name);
    input_node_names[i] = input_name;

    // print input node types
    Ort::TypeInfo type_info = session.GetInputTypeInfo(i);
    auto tensor_info = type_info.GetTensorTypeAndShapeInfo();

    ONNXTensorElementDataType type = tensor_info.GetElementType();
    printf("Input %d : type=%d\n", i, type);
 
    // print input shapes/dims
	input_node_dims.resize(2);
    input_node_dims[0] = 1;
	input_node_dims[1] = 10;
  }
  size_t input_tensor_size = 1 * 1 * 10;  // simplify ... using known dim values to calculate size
                                             // use OrtGetTensorShapeElementCount() to get official size!

  std::vector<int32_t> input_tensor_values(input_tensor_size);
  //std::vector<const char*> output_node_names = {"tag_predict/Reshape_1:0"};
  std::vector<const char*> output_node_names = {"encoder_kaggle/bert/pooler/dense/Tanh:0"};
  printf("here\n");

  // initialize input data with values in [0.0, 1.0]
  for (unsigned int i = 0; i < input_tensor_size; i++)
  {
	  input_tensor_values[i] = i+1;
  }
  // create input tensor object from data values
  auto memory_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
  std::vector<Ort::Value> ort_inputs;
  ort_inputs.emplace_back(Ort::Value::CreateTensor<int32_t>(memory_info, input_tensor_values.data(), input_tensor_size, input_node_dims.data(), input_node_dims.size()));
  ort_inputs.emplace_back(Ort::Value::CreateTensor<int32_t>(memory_info, input_tensor_values.data(), input_tensor_size, input_node_dims.data(), input_node_dims.size()));
  ort_inputs.emplace_back(Ort::Value::CreateTensor<int32_t>(memory_info, input_tensor_values.data(), input_tensor_size, input_node_dims.data(), input_node_dims.size()));

  auto output_tensors = session.Run(Ort::RunOptions{nullptr}, input_node_names.data(), ort_inputs.data(), ort_inputs.size(), output_node_names.data(), 1);
  

   printf(" TimeStart\n");
   struct timeval tstart,tend;
   double timeUsed;
   gettimeofday(&tstart,NULL);

   output_tensors = session.Run(Ort::RunOptions{nullptr}, input_node_names.data(), ort_inputs.data(), ort_inputs.size(), output_node_names.data(), 1);
   gettimeofday(&tend,NULL);
   timeUsed=1000000*(tend.tv_sec-tstart.tv_sec)+tend.tv_usec-tstart.tv_usec;
   printf(" Time=%f ms\n",timeUsed/1000);

Screenshots
Using Onnxruntime C++ API
Number of inputs = 3
Input 0 : name=bert_token_type_ids_Q:0
Input 0 : type=6
Input 1 : name=bert_input_mask_Q:0
Input 1 : type=6
Input 2 : name=bert_input_ids_Q:0
Input 2 : type=6
here
 TimeStart
 Time=599.910000 ms
Score for class [0] =  0.999141
Score for class [1] =  0.471896
Score for class [2] =  -0.960028
Score for class [3] =  -0.990269
Score for class [4] =  0.999590
Done!

Additional context
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Comment by tianleiwu** [comment]
@xiaoerqi, 

Could you try bert performance tools [here](https://github.com/microsoft/onnxruntime/tree/tlwu/bert_test_tools/onnxruntime/python/tools/bert)?

You can copy *.py file to the target machine, then use commands like:
```console
pip install onnxruntime onnx psutil
python bert_model_optimization.py --input bertmodel.onnx --output optimized_model.onnx --model_type bert_tf
python bert_perf_test.py --model optimized_model.onnx --batch_size 1 --sequence_length 128 --samples 100 --test_times 1 --inclusive
```
If your model is export by keras2onnx, use *--model_type bert_keras* in the second command.

If the result is better, you will need built the onnxruntime with OpenMP.

**Comment by xiaoerqi** [comment]
@tianleiwu ,
Thank you very much for your answer. After investigation, the server load is too high. Now the problem has been solved

**Comment by xiaoerqi** [comment]
@tianleiwu ,
Thank you very much for your answer. After investigation, the server load is too high. Now the problem has been solved

**Comment by dashesy** [comment]
The link to the tools (mentioned everywhere) are broken.
